### PR TITLE
Fix return types for matchers?

### PR DESCRIPTION
**Pull request by msutkowski**
I was testing out `extraReducers` with the matchers, and noticed that the return types were inaccurate. This is probably the wrong way to type this, but figured I'd kick it over.

I have a feeling this isn't the right solution though, either way.

### CODE REVIEW

File: examples/react/src/app/services/posts.ts [hunk 1]
```diff
@@ -30,19 +29,20 @@ const baseQuery = fetchBaseQuery({
...
     login: build.mutation<{ token: string; user: User }, any>({
       query: (credentials: any) => ({ url: 'login', method: 'POST', body: credentials }),
-      onSuccess: (arg, mutationApi, result) => {
-        const { token, user } = result;
-        // Set our user and token in the store for future requests
-        mutationApi.dispatch(setCredentials({ token, user }));
+      extraOptions: {
+        backoff: () => {
+          // We intentionally error once on login, and this breaks out of retrying. The next login attempt will succeed.
+          retry.fail({ fake: 'error' });
```
Comment (msutkowski) [inline examples/react/src/app/services/posts.ts:44]: This seemed like a good example for `retry.fail` usage.

**Review by msutkowski — COMMENTED** [review]
If you fix this, can you also add in an example of passing through errors to the rejected matcher? Or typed error usage in general?

File: examples/react/src/app/services/posts.ts (hunk 1 above)
Comment (phryneas) [reply examples/react/src/app/services/posts.ts:44]: Wouldn't that mean that we never retry, so the whole thing is pretty pointless?

Comment (msutkowski) [reply examples/react/src/app/services/posts.ts:44]: Kind of. Maybe this wasn't the best example 🤦 .

**Comment by phryneas** [comment]
Pushed some changes. From my side this looks ready, your call if we do any more changes to the example now, or some time later

**Pull request merged**